pkg/k8s/pipelineRun.go

```diff
@@ -208,72 +212,74 @@ func (r *pipelineRun) String() string {
+	// the call below returns the error from the function. This error is always nil. Hence there
+	// is no need to return that always-nil-error.
+	panicOnError(r.changeStatusAndStoreForRetry(func(s *api.PipelineStatus) (commitRecorderFunc, error) {
+		s.Result = result
 		now := metav1.Now()
-		r.apiObj.Status.FinishedAt = &now
-		return nil
-	})
+		s.FinishedAt = &now
+		return nil, nil
+	}))
 }
 
 // UpdateContainer ...
-func (r *pipelineRun) UpdateContainer(c *corev1.ContainerState) error {
+func (r *pipelineRun) UpdateContainer(c *corev1.ContainerState) {
 	if c == nil {
-		return nil
+		return
 	}
 	r.ensureCopy()
-	return r.changeStatusAndUpdateSafely(func() error {
-		r.apiObj.Status.Container = *c
-		return nil
-	})
+	panicOnError(r.changeStatusAndStoreForRetry(func(s *api.PipelineStatus) (commitRecorderFunc, error) {
+		s.Container = *c
+		return nil, nil
+	}))
 }
 
 // StoreErrorAsMessage stores the error as message in the status
 func (r *pipelineRun) StoreErrorAsMessage(err error, message string) error {
 	if err != nil {
 		text := fmt.Sprintf("ERROR: %s [%s]: %s", utils.Trim(message), r.String(), err.Error())
 		klog.V(3).Infof(text)
-		return r.UpdateMessage(text)
+		r.UpdateMessage(text)
 	}
 	return nil
 }
 
 // UpdateMessage stores string as message in the status
-func (r *pipelineRun) UpdateMessage(message string) error {
+func (r *pipelineRun) UpdateMessage(message string) {
 	r.ensureCopy()
 
-	return r.changeStatusAndUpdateSafely(func() error {
-		old := r.apiObj.Status.Message
+	panicOnError(r.changeStatusAndStoreForRetry(func(s *api.PipelineStatus) (commitRecorderFunc, error) {
+		old := s.Message
 		if old != "" {
-			his := r.apiObj.Status.History
+			his := s.History
 			his = append(his, old)
-			r.apiObj.Status.History = his
+			s.History = his
 		}
-		r.apiObj.Status.Message = utils.Trim(message)
-		r.apiObj.Status.MessageShort = utils.ShortenMessage(message, 100)
-		return nil
-	})
+		s.Message = utils.Trim(message)
+		s.MessageShort = utils.ShortenMessage(message, 100)
+		return nil, nil
+	}))
 }
 
 // UpdateRunNamespace overrides the namespace in which the builds happens
-func (r *pipelineRun) UpdateRunNamespace(ns string) error {
+func (r *pipelineRun) UpdateRunNamespace(ns string) {
 	r.ensureCopy()
-	return r.changeStatusAndUpdateSafely(func() error {
-		r.apiObj.Status.Namespace = ns
-		return nil
-	})
+	panicOnError(r.changeStatusAndStoreForRetry(func(s *api.PipelineStatus) (commitRecorderFunc, error) {
+		s.Namespace = ns
+		return nil, nil
+	}))
 }
 
 // UpdateAuxNamespace overrides the namespace hosting auxiliary services
 // for the pipeline run.
-func (r *pipelineRun) UpdateAuxNamespace(ns string) error {
+func (r *pipelineRun) UpdateAuxNamespace(ns string) {
 	r.ensureCopy()
-	return r.changeStatusAndUpdateSafely(func() error {
-		r.apiObj.Status.AuxiliaryNamespace = ns
-		return nil
-	})
+	panicOnError(r.changeStatusAndStoreForRetry(func(s *api.PipelineStatus) (commitRecorderFunc, error) {
```

Comment:
Why should we panic the whole program? Panic means a non recoverable error occurs.

Reply:
See https://github.com/SAP/stewardci-core/pull/248/files#r683193108

Reply:
Changed to the naming convention mustXXX as proposed by @rinckm .
In that cases the called function returns an error, but from the way we call the function we know that with this invocation the returned error will be always nil. In case our assumtion is false the program will panic. This is better than just swallowing since when we swallow we don't have the awarness that something is wrong.

Reply:
@jia-jerry: do you have concern wrt the `mustXXX` pattern?